src/mono/mono/mini/interp/interp-internals.h

```diff
@@ -101,6 +102,10 @@ typedef enum {
 
 #define PROFILE_INTERP 0
 
+#if !HOST_BROWSER && __GNUC__
```

Comment:
```suggestion
#if !HOST_BROWSER && !HOST_WASI && __GNUC__
```

Reply:
The real reason for disabling on browser is not because this PR has some problems with wasm, but because this PR leads to jiterpreter trace aborts since support is not yet added. On WASI there is no jiterpreter support so this PR should still have perf gains.